step-capstone/src/components/Firebase/firebase.js

```diff
@@ -69,6 +72,51 @@ class Firebase {
     return tripListRef.add(trip);
   }
 
+  async deleteTrip(reference) {
+    const tripRef = this.db.doc(reference);
+    return new Promise((resolve, reject) =>
+     {
+      tripRef.delete().then(() => {
+        console.log("Trip successfully deleted!");
+        resolve();
+      }).catch(function(error) {
+          console.error("Error removing trip: ", error);
+          reject()
+      })
```

Comment:
I think you can remove this promise here. if you just return tripRef.delete() it will return a promise to its caller. Then then and catch should be handled by the caller, because this firebase function does not know what needs to happen after the delete succeeds or fails.

Reply:
good point

Reply:
addressed